plugins/catalog-backend/src/service/AuthorizedEntitiesCatalog.ts

```diff
@@ -70,8 +72,84 @@ export class AuthorizedEntitiesCatalog implements EntitiesCatalog {
     return this.entitiesCatalog.removeEntityByUid(uid);
   }
 
-  entityAncestry(entityRef: string): Promise<EntityAncestryResponse> {
-    // TODO: Implement permissioning
-    return this.entitiesCatalog.entityAncestry(entityRef);
+  async entityAncestry(
+    entityRef: string,
+    options?: { authorizationToken?: string },
+  ): Promise<EntityAncestryResponse> {
+    const rootEntityAuthorizeResponse = (
+      await this.permissionApi.authorize(
+        [{ permission: catalogEntityReadPermission, resourceRef: entityRef }],
+        { token: options?.authorizationToken },
+      )
+    )[0];
+    if (rootEntityAuthorizeResponse.result === AuthorizeResult.DENY) {
+      throw new NotAllowedError();
```

Comment:
What does this API do if the entityRef doesn't exist? Wondering if this should perhaps be a NotFoundError?

Reply:
If I'm understanding you correctly, [yes](https://github.com/backstage/backstage/pull/8805#discussion_r782512182).